Class.cpp

```diff
@@ -0,0 +1,207 @@
+// System includes
+#include <iterator>
+
+// Local includes
+#include "Class.h"
+
+// Default constructor
+Class::Class()
+{
+}
+
+// Constructor given a name
+Class::Class(const std::string &name)
+	:mName(name)
+{
+}
+
+// Changes the name of an existing class to a new given name
+void Class::rename(const std::string &newName)
+{
+    mName = newName;
+}
+
+// Returns the name of a class
+const std::string &Class::name() const
+{
+    return mName;
+}
+
+// Returns the raw name of a class
+const char *Class::rawName() const
+{
+    return mName.c_str();
+}
+
+// Adds a field to the classes' field vector
+void Class::addField(const Field &field)
+{
+    mFields.push_back(field);
+}
+
+// Adds a method to the classes' method vector
+void Class::addMethod(const Method &method)
+{
+    mMethods.push_back(method);
+}
+
+// Removes a field from the vector of fields at the given index
+void Class::removeField(int index)
+{
+	std::list<Field>::iterator it = mFields.begin();
+	std::advance(it, index);
+	mFields.erase(it);
+}
+
+// Removes a field from the vector of fields after finding its index
+void Class::removeField(const std::string &name)
+{
+    unsigned int index = 0;
+    for (auto &i : mFields)
+    {
+        if (i.name() == name)
+        {
+			mFields.remove(i);
+        }
+    }
+}
+
+// Removes a method from the vector of methods at the given index
+void Class::removeMethod(int index)
+{
+	std::list<Method>::iterator it = mMethods.begin();
+	std::advance(it, index);
+	mMethods.erase(it);
+}
+
+// Removes a method from the vector of methods after finding its index
+void Class::removeMethod(const std::string &name)
+{
+	std::string temp;
+
+    for (auto &i : mMethods)
+    {
+		temp = i.name();
+        if (temp == name)
+        {
+			mMethods.remove(i);
+        }
+    }
+}
+
+// Returns a method from the vector of methods given its name
+Method &Class::method(const std::string &name)
+{
+	Method* result = nullptr;
+	for (auto& i : mMethods)
+	{
+		if (i.name() == name)
+			result = &i;
```

Comment:
you can just return `&i` here and change the return to nullptr below (result is unnecessary)